zetaclient/zetacore/broadcast.go

```diff
@@ -107,6 +111,16 @@ func (c *Client) Broadcast(
 		return "", errors.Wrap(err, "unable to encode tx")
 	}
 
+	// skip broadcasting if mempool is already congested
+	resp, err := c.cometBFTClient.NumUnconfirmedTxs(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get unconfirmed tx count")
+	}
+
+	if resp.Count >= unconfirmedTxsThreshold {
+		return "", fmt.Errorf("mempool is congested, unconfirmed tx count is %d", resp.Count)
```

Comment:
If the mempool is congested, we know no broadcast will work at all, at this point it seems to make more senseto just pause observation in the zetaclient

Reply:
So you would prefer add a single process to monitor the # of unconfirmed txs in mempool, and this global value is used by every chain to pause their own goroutines?

Reply:
https://github.com/zeta-chain/node/pull/4291/commits/274eafa116b2830ae37c6da665c07e5149dd0364